app/views/edit/build-config.html

```diff
@@ -536,7 +534,99 @@ <h3 class="with-divider">Run Policy
                         <div class="help-block" ng-switch-default>Builds triggered from this Build Configuration will run using the {{updatedBuildConfig.spec.runPolicy | sentenceCase}} policy.</div>
                       </div>
                     </div>
-                  
+
+                    <div class="section" ng-if="view.advancedOptions">
+                      <div ng-if="!(updatedBuildConfig | isJenkinsPipelineStrategy)" class="section">
+                        <h3 class="with-divider">Build Hooks
+                          <span class="help action-inline">
+                            <a href>
+                              <i class="pficon pficon-help" data-toggle="tooltip" aria-hidden="true" data-original-title="Build hooks allow behavior to be injected into the build process."></i>
+                            </a>
+                          </span>
+                        </h3>
+
+                        <div>
+                          <div class="form-group">
+                            <h4>Hook Types</h4>
+                            <ui-select required
+                                       ng-model="buildHookSelection.type"
+                                       title="Choose a Build Hook type to edit">
+                              <ui-select-match>{{$select.selected | sentenceCase}}</ui-select-match>
+                              <ui-select-choices repeat="type in buildHookTypes">
+                                {{type | sentenceCase }}
+                              </ui-select-choices>
+                            </ui-select>
+                          </div>
+
+                          <fieldset>
+                            <div ng-show="buildHookSelection.type === 'Shell Script'">
+                              <h4>Script</h4>
+                              <div
+                                ui-ace="{
+                                  mode: 'yaml',
+                                  theme: 'eclipse',
+                                  onLoad: aceLoaded,
+                                  rendererOptions: {
+                                    fadeFoldWidgets: true,
+                                    showPrintMargin: false
+                                  }
+                                }"
+                                ng-model="buildConfig.spec.postCommit.script"
+                                class="ace-bordered ace-read-only ace-inline dockerfile-mode mar-top-md">
+                              </div>
+                            </div>
+                            <div ng-show="buildHookSelection.type === 'Command'">
+                              <h4>Command</h4>
+                              <edit-command
+                                args="buildConfig.spec.postCommit.command"
+                                isRequired="true">
```

Comment:
I'm not sure setting `isRequired="true"` is correct since this is inside an `ng-show` block, not an `ng-if`. If I don't have a command and switch types, can I submit the form?